src/components/PupilComponents/LessonEngineProvider/LessonEngineProvider.test.tsx

```diff
@@ -60,131 +62,83 @@ describe("LessonEngineProvider", () => {
 
     act(() => {
       result.current.completeSection("intro");
-    });
-    expect(result.current.completedSections).toEqual(["intro"]);
-
-    act(() => {
       result.current.proceedToNextSection();
     });
     expect(result.current.currentSection).toEqual("starter-quiz");
   });
 
-  it("tracks completed sections", () => {
+  it("returns to the first section when proceedToNextSection is called when all sections are complete", () => {
     const { result } = renderHook(() => useLessonEngineContext(), {
-      wrapper: (props) =>
-        providerWrapper({
-          ...props,
-        }),
+      wrapper: ProviderWrapper,
     });
 
     if (result.current === null) {
       throw new Error("result.current is null");
     }
 
-    expect(result.current.completedSections).toEqual([]);
-
-    const { completeSection } = result.current;
-
-    lessonSections.forEach((s, i) => {
-      act(() => {
-        completeSection(s);
+    act(() => {
+      allLessonReviewSections.forEach((section) => {
+        result.current.completeSection(section);
       });
-
-      expect(result.current.completedSections).toEqual(
-        lessonSections.slice(0, i + 1),
-      );
+      result.current.proceedToNextSection();
     });
+
+    expect(result.current.currentSection).toEqual(allLessonReviewSections[0]);
   });
 
   it("returns to overview on completeSection when not all sections are complete", () => {
     const { result } = renderHook(() => useLessonEngineContext(), {
-      wrapper: (props) =>
-        providerWrapper({
-          ...props,
-        }),
+      wrapper: ProviderWrapper,
     });
 
     if (result.current === null) {
       throw new Error("result.current is null");
     }
 
-    ["intro", "starter-quiz", "video"].forEach((s) => {
-      act(() => {
-        result.current.completeSection(s as LessonSection);
-      });
-
+    allLessonReviewSections.forEach((s) => {
       expect(result.current.currentSection).toEqual("overview");
-    });
 
-    act(() => {
-      result.current.completeSection("exit-quiz");
+      act(() => {
+        result.current.completeSection(s);
+      });
     });
 
     expect(result.current.currentSection).toEqual("review");
   });
 
-  it("doesn't duplicate completed sections", () => {
+  it("sets `isComplete` for the section when it is completed", () => {
```

Comment:
If writing tests targeting a the reducer i'd probably do something like this:

```typescript
describe('completeSection', () => {
  const newState = lessonEngineReducer({
    currentSection: 'starter-quiz',
    lessonReviewSections: allLessonReviewSections,
    sections: {}
  }, { type: 'completeSection', section: 'starter-quiz' });

  it('sets `isComplete` for the section', () => {
    expect(newState.sections['starter-quiz']?.isComplete).toEqual(true);
  });

  it('returns to the overview section'
    expect(newState.currentSection).toEqual('overview');
  });
});
```

I think it's much easier to follow this as documentation without all the testing library cruft muddying the way